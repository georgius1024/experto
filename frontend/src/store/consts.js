const consts = {
  PUBLICATION_CAMERA_ADD: 'publication-add-camera',
  PUBLICATION_CAMERA_REMOVE: 'publication-remove-camera',
  PUBLICATION_SCREEN_ADD: 'publication-add-screen',
  PUBLICATION_SCREEN_REMOVE: 'publication-remove-screen',
  PUBLICATION_REMOVE_ALL: 'publication-remove-all',
  SUBSCRIPTION_ADD: 'subscription-add',
  SUBSCRIPTION_REMOVE: 'subscription-remove',
  SUBSCRIPTIONS_REMOVE_ALL: 'subscription-remove-all',
  SUBSCRIPTIONS_UPDATE_ALL: 'subscription-update-all',
  CAMERA_AUDIO_ENABLE: 'camera-audio-enable',
  CAMERA_AUDIO_DISABLE: 'camera-audio-disable',
  CAMERA_VIDEO_ENABLE: 'camera-video-enable',
  CAMERA_VIDEO_DISABLE: 'camera-video-disable',
  SCREEN_VIDEO_ENABLE: 'screen-video-enable',
  SCREEN_VIDEO_DISABLE: 'screen-video-disable',
}

export default consts
