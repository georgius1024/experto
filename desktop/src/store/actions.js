import consts from './consts'

function actionFactory(type) {
  return function(payload) {
    return {
      type,
      payload
    }
  }
}
const actions = {
  register: actionFactory(consts.REGISTER),
  unregister: actionFactory(consts.UNREGISTER),
  refresh: actionFactory(consts.REFRESH),
  publicationCameraAdd: actionFactory(consts.PUBLICATION_CAMERA_ADD),
  publicationCameraRemove: actionFactory(consts.PUBLICATION_CAMERA_REMOVE),
  publicationScreenAdd: actionFactory(consts.PUBLICATION_SCREEN_ADD),
  publicationScreenRemove: actionFactory(consts.PUBLICATION_SCREEN_REMOVE),
  publicationRemoveAll: actionFactory(consts.PUBLICATION_REMOVE_ALL),
  subscriptionAdd: actionFactory(consts.SUBSCRIPTION_ADD),
  subscriptionRemove: actionFactory(consts.SUBSCRIPTION_REMOVE),
  subscriptionsRemoveAll: actionFactory(consts.SUBSCRIPTIONS_REMOVE_ALL),
  subscriptionsUpdateAll: actionFactory(consts.SUBSCRIPTIONS_UPDATE_ALL),
  cameraAudioEnable: actionFactory(consts.CAMERA_AUDIO_ENABLE),
  cameraAudioDisable: actionFactory(consts.CAMERA_AUDIO_DISABLE),
  cameraVideoEnable: actionFactory(consts.CAMERA_VIDEO_ENABLE),
  cameraVideoDisable: actionFactory(consts.CAMERA_VIDEO_DISABLE),
  screenVideoEnable: actionFactory(consts.SCREEN_VIDEO_ENABLE),
  screenVideoDisable: actionFactory(consts.SCREEN_VIDEO_DISABLE),
}

export default actions
