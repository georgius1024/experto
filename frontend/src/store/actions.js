import consts from './consts'

function actionFactory(type) {
  return function(payload) {
    return {
      type,
      payload
    }
  }
}
const actions = {
  publicationAdd: actionFactory(consts.PUBLICATION_ADD),
  publicationRemove: actionFactory(consts.PUBLICATION_REMOVE),
  subscriptionAdd: actionFactory(consts.SUBSCRIPTION_ADD),
  subscriptionRemove: actionFactory(consts.SUBSCRIPTION_REMOVE),
  subscriptionRemoveAll: actionFactory(consts.SUBSCRIPTION_REMOVE_ALL),
  subscriptionUpdateAll: actionFactory(consts.SUBSCRIPTION_UPDATE_ALL),
  audioEnable: actionFactory(consts.AUDIO_ENABLE),
  audioDisable: actionFactory(consts.AUDIO_DISABLE),
  videoEnable: actionFactory(consts.VIDEO_ENABLE),
  videoDisable: actionFactory(consts.VIDEO_DISABLE),
  userNameUpdate: actionFactory(consts.USERNAME_UPDATE),
  appointmentUpdate: actionFactory(consts.APPOINTMENT_UPDATE),
  controlSocketUpdate: actionFactory(consts.CONTROL_SOCKET_UPDATE)
}

export default actions
