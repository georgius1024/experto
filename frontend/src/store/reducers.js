import consts from './consts'
const reducers = {
  publicationsReducer(state = { camera: null, screen: null }, action) {
    switch (action.type) {
      case consts.PUBLICATION_CAMERA_ADD:
        return { ...state, camera: action.payload }
      case consts.PUBLICATION_CAMERA_REMOVE:
        return { ...state, camera: null }
      case consts.PUBLICATION_SCREEN_ADD:
        return { ...state, screen: action.payload }
      case consts.PUBLICATION_SCREEN_REMOVE:
        return { ...state, screen: null }
      case consts.PUBLICATION_REMOVE_ALL:
        return { camera: null, screen: null }
      default:
        return state
    }
  },
  subscriptionsReducer(state = [], action) {
    switch (action.type) {
      case consts.SUBSCRIPTION_ADD:
        return [...state.filter(subscription => subscription.channel !== action.payload.channel), action.payload]
      case consts.SUBSCRIPTION_REMOVE:
        return state.filter(subscription => subscription.channel !== action.payload)
      case consts.SUBSCRIPTIONS_REMOVE_ALL:
        return []
      case consts.SUBSCRIPTIONS_UPDATE_ALL:
        return action.payload
      default:
        return state
    }
  },
  cameraAudioReducer(state = true, action) {
    switch (action.type) {
      case consts.CAMERA_AUDIO_ENABLE:
        return true
      case consts.CAMERA_AUDIO_DISABLE:
        return false
      default:
        return state
    }
  },
  cameraVideoReducer(state = true, action) {
    switch (action.type) {
      case consts.CAMERA_VIDEO_ENABLE:
        return true
      case consts.CAMERA_VIDEO_DISABLE:
        return false
      default:
        return state
    }
  },
  screenVideoReducer(state = true, action) {
    switch (action.type) {
      case consts.SCREEN_VIDEO_ENABLE:
        return true
      case consts.SCREEN_VIDEO_DISABLE:
        return false
      default:
        return state
    }
  }
}

export default reducers
