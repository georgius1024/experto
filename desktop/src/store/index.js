import { combineReducers, createStore } from 'redux'
import reducers from './reducers'

const mainReducer = combineReducers({
  publications: reducers.publicationsReducer,
  subscriptions: reducers.subscriptionsReducer,
  cameraAudio: reducers.cameraAudioReducer,
  cameraVideo: reducers.cameraVideoReducer,
  screenVideo: reducers.screenVideoReducer,
  userName: reducers.usernameReducer,
  appointment: reducers.appointmentReducer,
  controlSocket: reducers.controlSocketReducer
})

const store = createStore(mainReducer)

export default store
