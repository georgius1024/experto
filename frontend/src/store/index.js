import { combineReducers, createStore } from 'redux'
import reducers from './reducers'

const mainReducer = combineReducers({
  publication: reducers.publicationReducer,
  subscriptions: reducers.subscriptionsReducer,
  audio: reducers.audioReducer,
  video: reducers.videoReducer,
  userName: reducers.usernameReducer,
  appointment: reducers.appointmentReducer,
  controlSocket: reducers.controlSocketReducer
})

const store = createStore(mainReducer)

export default store
