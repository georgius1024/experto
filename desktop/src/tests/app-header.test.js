import React from 'react'
import renderer from 'react-test-renderer'
import Component from '../components/app-header'
import { Provider } from 'react-redux'
import store from '../store'
import { MemoryRouter } from 'react-router-dom'
describe('App header component', () => {
  test('it matches the snapshot', () => {
    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <Component />
        </MemoryRouter>
      </Provider>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
