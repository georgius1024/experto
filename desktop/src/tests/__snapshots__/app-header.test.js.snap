// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App header component it matches the snapshot 1`] = `
<nav
  className="navbar navbar-expand navbar-dark bg-primary sticky-top"
>
  <a
    className="navbar-brand"
    href="#"
    onClick={[Function]}
  >
    Experto Crede
  </a>
  <span
    className="ml-auto mr-2 navbar-text"
  >
    Эксперт не зарегистрирован
  </span>
</nav>
`;
